pkg/asset/cluster/tfvars/tfvars.go

```diff
@@ -508,8 +508,9 @@ func (t *TerraformVariables) Generate(ctx context.Context, parents asset.Parents
 		privateZoneName := ""
 
 		if installConfig.Config.GCP.UserProvisionedDNS != dns.UserProvisionedDNSEnabled {
+			project := manifests.GetProjectForDNSZones(installConfig)
 			if installConfig.Config.Publish == types.ExternalPublishingStrategy {
-				publicZone, err := client.GetDNSZone(ctx, installConfig.Config.GCP.ProjectID, installConfig.Config.BaseDomain, true)
+				publicZone, err := client.GetDNSZone(ctx, project, installConfig.Config.BaseDomain, true)
```

Comment:
Hm it seems this is setting the public zone off of the new field, but I think that new field should only be for private zones. Am I missing something?

Reply:
I think I went off of the fact that the private and public zone would exist in the same project. It is definitely possible that they shouldn't. I thought that was in the epic or some notes, but I may be wrong.